InformationScripting/src/visualization/DefaultVisualizer.cpp

```diff
@@ -111,13 +115,35 @@ void DefaultVisualizer::visualize(const TupleSet& ts)
 			while (nodeVisualizationIt != Visualization::Item::nodeItemsMap().end() && nodeVisualizationIt.key() == node)
 			{
 				auto item = *nodeVisualizationIt++;
-				auto overlay = new Visualization::SelectionOverlay(
-							item, Visualization::SelectionOverlay::itemStyles().get(it.value() + "Highlight"));
+				auto style = createSelectionStyle(it.value());
+				auto overlay = new Visualization::SelectionOverlay(item, style);
 				item->addOverlay(overlay, HIGHLIGHT_OVERLAY_GROUP);
 			}
 		}
 	}
 }
 
+Visualization::ItemStyle* DefaultVisualizer::createSelectionStyle(const QString& color)
+{
+	auto style = Visualization::StyleLoader().loadStyle<Visualization::ItemStyle>("item/SelectionOverlay",
```

Comment:
This method will end up modifying the "redHighlight" style, which will be unexpected for other items that use it. Styles are meant to be readonly. Therefore, you should rather construct a new style using a constructor.

Though another problem is that styles are only ever destroyed at program exit. So if you generate a new style every time a different color is supplied, this will leak significant amounts of memory.

Typically, the way I deal with dynamic visualization properties is to make these part of the item. In your particular case you should look at (and override) Item::customShapeColor()